session/session.go

```diff
@@ -1322,6 +1322,8 @@ func (s *session) RefreshTxnCtx(ctx context.Context) error {
 		return errors.Trace(err)
 	}
 
+	s.GetSessionVars().TxnCtx.Binlog = nil
```

Comment:
Is this root cause of OOM?

Reply:
I think this is the reason.

Reply:
We should clear the TxnCtx in `NewTxn`.

Reply:
Then, we needn't to change the line of 825 in `server/conn.go`.